frontend/src/screens/wallet/receive/ReceiveOnchain.tsx

```diff
@@ -0,0 +1,425 @@
+import { copyToClipboard } from "src/lib/clipboard";
+import { cn } from "src/lib/utils";
+import { MempoolUtxo, SwapResponse } from "src/types";
+
+import TickSVG from "public/images/illustrations/tick.svg";
+import Lottie from "react-lottie";
+import { Link, useNavigate } from "react-router-dom";
+import FormattedFiatAmount from "src/components/FormattedFiatAmount";
+import { Alert, AlertDescription, AlertTitle } from "src/components/ui/alert";
+import { Input } from "src/components/ui/input";
+import { Label } from "src/components/ui/label";
+import { LoadingButton } from "src/components/ui/loading-button";
+import { useTheme } from "src/components/ui/theme-provider";
+import { MIN_SWAP_AMOUNT } from "src/constants";
+import { useBalances } from "src/hooks/useBalances";
+import { useSwapFees } from "src/hooks/useSwaps";
+import { request } from "src/utils/request";
+
+export default function ReceiveOnchain() {
+  const [receiveType, setReceiveType] = useState("onchain");
+
+  useEffect(() => {
+    const queryParams = new URLSearchParams(location.search);
+    const receiveType = queryParams.get("type");
+    if (receiveType) {
+      setReceiveType(receiveType);
+    }
+  }, []);
+
+  return (
+    <div className="grid gap-5">
+      <AppHeader title="Receive On-chain" />
+      <div className="w-full max-w-lg grid gap-5">
+        <div className="flex items-center text-center text-foreground font-medium rounded-xl bg-muted p-1">
+          <div
+            className={cn(
+              "cursor-pointer rounded-lg flex-1 py-1.5 text-sm",
+              receiveType == "onchain" &&
+                "text-foreground bg-background shadow-md"
+            )}
+            onClick={() => setReceiveType("onchain")}
+          >
+            Receive to On-chain
+          </div>
+          <div
+            className={cn(
+              "cursor-pointer rounded-lg flex-1 py-1.5 text-sm",
+              receiveType == "spending" &&
+                "text-foreground bg-background shadow-md"
+            )}
+            onClick={() => setReceiveType("spending")}
+          >
+            Receive to Spending
+          </div>
+        </div>
+        {receiveType == "onchain" ? (
+          <ReceiveToOnchain />
+        ) : (
+          <ReceiveToSpending />
+        )}
+      </div>
+    </div>
+  );
+}
+
+function ReceiveToOnchain() {
+  const { data: onchainAddress, getNewAddress } = useOnchainAddress();
+  const { toast } = useToast();
+  const { data: mempoolAddressUtxos } = useMempoolApi<MempoolUtxo[]>(
+    onchainAddress ? `/address/${onchainAddress}/utxo` : undefined,
+    3000
+  );
+
+  const [txId, setTxId] = useState("");
+  const [confirmedAmount, setConfirmedAmount] = useState<number | null>(null);
+  const [pendingAmount, setPendingAmount] = useState<number | null>(null);
+
+  useEffect(() => {
+    if (!mempoolAddressUtxos || mempoolAddressUtxos.length === 0) {
+      return;
+    }
+
+    if (txId) {
```

Comment:
all this code is duplicated from the deposit page right? :thinking:

Reply:
Yup, it's the same code, just the UI is a bit changed